### DISCUSSION TITLE
Request for an Example of Assertion Usage in Device Service Development

### DISCUSSION THREAD

**Discussion by bighb69738** [edgexfoundry/community#322] [Q&A]
Hello,

I am currently developing a device service and would like to implement Assertions to validate the readings of device resources. According to the official documentation, Assertions are used to check if the device readings meet certain conditions, and if they fail, an error is returned, and the device's state is set to DISABLED.

However, I am having trouble finding a comprehensive example on how to properly set up Assertions, specifically in the following areas:

How to set up range checks (e.g., checking if temperature is within a certain range).
How to handle error messages and statuses when an Assertion fails.
A full example of configuring Assertion in the PropertyValue of a device resource.
I noticed that the documentation doesn't provide full examples, and I would appreciate it if you could share some concrete examples of how to apply Assertions in real-world device service development.

Thank you!

**Comment by jiekechoo** [comment]
@bighb69738 
It seems that you are working on a workflow or pipeline liked system.  Is that right?
You could try the `Rules Engine`: [eKuiper](https://docs.edgexfoundry.org/3.1/microservices/support/eKuiper/Ch-eKuiper/).

**Reply by bighb69738** [reply]
Hi @jiekechoo ,
I just want to know how to use the Assertion attribute in EdgeX's profile. Are there any related examples?

**Reply by jiekechoo** [reply]
@bighb69738 Please refer the [Assertions and Mappings](https://docs.edgexfoundry.org/3.1/design/adr/device-service/0011-DeviceService-Rest-API/#assertions-and-mappings) ADR.
